### DISCUSSION TITLE
QA testing, illuminance values out of bounds

### DISCUSSION THREAD

**Discussion by jinnatar** [rbaron/b-parasite#143] [Q&A]
I've got my first run of 5 soldered together (mini hot plate reflowing the nRF on, super fun.) .. Most of them are performing well but one of them is getting wild illuminance values, such as `11,687.0 lx`. It's not a static faulty value, another reading is `11,746.0 lx`. In the meanwhile it's buddies are getting around `100 lx`. Two are also getting a reading of `0 lx`. So in total 2 out of 5 are stuck at zero, 2 get sane values and one has some sneaky direct sunlight going on. All other sensors seem sane though so I seem to be failing just the one.

I'm assuming it's just a shoddy joint on one of the nRF pins, but anyone troubleshot already deeper into which one(s) would cause these kinds of results? Trying to learn to spot bad joints so I can do quick QA faster and learn better how much solder paste to apply. Would be nice to have a pinout picture in the docs so those of us not versed well in KiCAD could see it as well.

**Comment by rbaron** [comment]
I think there are two points:
1. The sensivity of the sensor (i.e.: some reading 0 lx, some reading 100 lx under similar conditions of (I assume) indirect indoor lighting). When designing the [ALS-PT19](https://www.mouser.ch/ProductDetail/Everlight/ALS-PT19-315C-L177-TR8) circuit, we need to choose between precision in a small range or wide range. I leaned toward wide range, so we can measure more or less from total dark (0 lx) to direct sunlight (100k lx). It means a 100 lx difference is "meaningless".
2. Faulty sensor. It could be a number of things. A bad solder join like you suggested (you can check the schematic or the board layout for usual suspects). But these are kinda finicky to manual solder. I once ordered a batch with all of them in the wrong orientation and ruined ~50% of them trying to fix it (but eventually got all working with new phototransistors). You can see on the datasheet that the recommended max soldering temp is relatively low:

<img width="856" alt="image" src="https://github.com/rbaron/b-parasite/assets/1573409/0772a273-e726-4a07-aa8f-50c3b8c35170">

That being said, my suggestion would be to just replace it with a fresh one.

**Reply by jinnatar** [reply]
Yeah, I'm not worried about one saying 50 and one 100, that's totally fine. But sounds like I need to redo both the constant-zero and constant daylight ones. I'm doing the reflow at 170C so damaging the sensors shouldn't be a huge issue.

Thanks for the tips, I'll fiddle a bit more and see if they're rescuable.

**Comment by Judman** [comment]
I just ran into a similar issue with 1 of 5 boards I got from JCLPCB. After looking at this thread I went to touch up the solder on the ALS-PT19 thinking it might be a bad joint.

Turns out the sensor was fine, but the R10 resistor was out of position and causing light values of 16k.

Side by side, two boards from the factory. The left board worked as expected, the right one has the 15k in a dark room.
![20230711_102440](https://github.com/rbaron/b-parasite/assets/3705824/9c00a9f3-6006-4353-918e-2278125dbdec)

Fixed. Holy cow these tiny parts are insane to hand solder. They stick to everything!
![20230711_104902](https://github.com/rbaron/b-parasite/assets/3705824/c8255d7b-2660-4499-8c95-9290c93c65de)
